pkg/appfs/copier.go

```diff
@@ -147,6 +157,14 @@ func NewAferoCopier(fs afero.Fs) Copier {
 	return &aferoCopier{fs: fs}
 }
 
+func (f *aferoCopier) Exist(slug, version, shasum string) (bool, error) {
+	appDir := path.Join("/", slug, version)
+	if shasum != "" {
+		appDir += "-" + shasum
+	}
+	return afero.DirExists(f.fs, appDir)
+}
+
 func (f *aferoCopier) Start(slug, version, shasum string) (bool, error) {
 	f.appDir = path.Join("/", slug, version)
```

Comment:
You could use `f.Exist()` here, couldn't you?

Reply:
`f.appDir` is not defined in `f.Exist()` and it is used later to create the directory (`dir := path.Dir(f.appDir)`).

Reply:
Then why not assigning the path of the directory to f.appDir in Exist like is done for the Swift copier?